src/display/annotation_layer.js

```diff
@@ -289,13 +289,15 @@ class AnnotationElement {
 }
 
 class LinkAnnotationElement extends AnnotationElement {
-  constructor(parameters) {
-    const isRenderable = !!(
+  constructor(parameters, isRenderable = false) {
+    const _isRenderable = !!(
       parameters.data.url ||
       parameters.data.dest ||
-      parameters.data.action
+      parameters.data.action ||
+      isRenderable
```

Comment:
Why is this necessary though; couldn't you simply add `parameters.data.isTooltipOnly` instead here, since that will only (possibly) be defined for the "PushButton pretending to be LinkAnnotation"-case as far as I can tell?

This way, you'd reduce the amount of constructor-related changes in this file.

Reply:
I did that in the first version but it's quite confusing to have in LinkAnnotation (display) something defined in a WidgetAnnotation (core) so I moved the stuff in PushButton for this reason.
Anyway I'm fine to change it back... just tell me what you prefer.

Reply:
@Snuffleupagus thoughts?

Reply:
> @Snuffleupagus thoughts?

Sorry, this PR slipped my mind!

With the variable re-name reverted at https://github.com/mozilla/pdf.js/pull/12333/files#diff-1097e3b131ba22f8e8a6c2f20e3ab5ecR293, I honestly think that this is fine as-is :-)

Reply:
> @Snuffleupagus thoughts?

Sorry, this PR slipped my mind!

With the variable name changed back to `isRenderable` at https://github.com/mozilla/pdf.js/pull/12333/files#diff-1097e3b131ba22f8e8a6c2f20e3ab5ecR293, I honestly think that this is fine as-is :-)